### PR TITLE
Fix/pylint issues

### PR DESCRIPTION
**Pull request by hunterpaulson**
## F´ Pull Request
|**Field**|**Value**|
|:---|:---|
|**_Submission Date_**| 2020-07-07 |
|**_Submitter_**| @hunterpaulson |
|**_Originating Project/Creator_**| fprime |
|**_Base Branch_**| devel |
|**_Short Description_**| Fixes python issues highlighted by pylint |
|**_Effected Component_**| Fw/Python and Gds |
|**_Effected Architectures(s)_**| Posix |
|**_Related Issue(s)_**| n/a |
|**_Has Unit Tests (y/n)_**| y |
|**_Build Checked (y/n)_**| y |
|**_Unit Tests Run (y/n)_**| y |
|**_Documentation Included (y/n)_**| n/a |

---
## Change Description
Fixes hundreds of little pylint errors and warnings within the fprime python code base. Pretty much every python file within Fw/Python and the Gds had some sort of warning. Many of these were wildcard import issues and naming issues where variables were redefining private ones.
## Rationale
We are trying to improve our python code quality. Currently our python code has the second lowest rating of D on [lgtm](https://lgtm.com/projects/g/nasa/fprime/). Meaning that we are in the bottom 5-25 percentile when compared to projects of similar size.
## Testing Recommendations
A couple new pytests were created and updated so you can run those as well as anything else that runs python code.
## Future Work
This fixes a majority of the original 902 issues. However there are still many more that should be resolved.

### CODE REVIEW

File: Fw/Python/src/fprime/common/models/serialize/enum_type.py [hunk 1]
```diff
@@ -126,16 +133,4 @@ def __repr__(self):
...
-        val = EnumType("SomeEnum", members, "MEMB3")
-        print("Value: %s" % val.val)
-        buff = val.serialize()
-        type_base.showBytes(buff)
-        val2 = EnumType("SomeEnum", members)
-        val2.deserialize(buff, len(buff))
-        print("Deserialize: %s" % val2.val)
-    except TypeException as e:
-        print("Exception: %s" % e.getMsg())
+    pass
```
Comment (LeStarch) [inline Fw/Python/src/fprime/common/models/serialize/enum_type.py:136]: Same here, and anywhere else.

```
if __name__ == "__main__":
    pass
```

Does nothing at all.

File: Fw/Python/src/fprime/common/models/serialize/time_type.py [hunk 2]
```diff
@@ -19,14 +19,15 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
-import time
-from datetime import *
-from pytz import *
+import datetime
+
+# from pytz import TimeBase
```
Comment (LeStarch) [inline Fw/Python/src/fprime/common/models/serialize/time_type.py:24]: Why is this commented out?

File: Fw/Python/src/fprime/fbuild/cmake.py [hunk 3]
```diff
@@ -461,7 +460,10 @@ def _run_cmake(
         :return: (stdout, stderr)
         Note: !!! this function has potential File System side-effects !!!
         """
-        cm_environ = copy.copy(os.environ)
+        if environment is None:
```
Comment (LeStarch) [inline Fw/Python/src/fprime/fbuild/cmake.py:463]: Ugh.  I hate this fact about python.  It is the correct fix.

File: Fw/Python/src/fprime/fbuild/cmake.py [hunk 4]
```diff
@@ -536,7 +538,7 @@ def _communicate(proc, stdout, stderr, print_output=True):
                     line = key.fileobj.readline().decode().replace("\r\n", "\n")
                 # Some systems (like running inside Docker) raise an io error instead of returning "" when the device
                 # is ended. Not sure why this is, but the effect is the same, on IOError assume end-of-input
-                except IOError as ioe:
+                except IOError as dummy_ioe:
```
Comment (LeStarch) [inline Fw/Python/src/fprime/fbuild/cmake.py:541]: Just remove the variable.  You can also use "_" to discard unused variable.

File: Fw/Python/test/fprime/common/models/serialize/test_types.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
+"""
```
Comment (LeStarch) [inline Fw/Python/test/fprime/common/models/serialize/test_types.py:1]: UNIT TESTS!!!!!! HURRAY!

File: Gds/src/fprime_gds/common/adapters/base.py [hunk 6]
```diff
@@ -22,11 +22,9 @@ class BaseAdapter(abc.ABC):
 
     def open(self):
         """Null default implementation """
-        pass
```
Comment (LeStarch) [inline Gds/src/fprime_gds/common/adapters/base.py:25]: Is this legal python?

File: Gds/src/fprime_gds/common/adapters/ip.py [hunk 7]
```diff
@@ -46,7 +46,9 @@ class IpAdapter(fprime_gds.common.adapters.base.BaseAdapter):
     KEEPALIVE_INTERVAL = (
         0.500  # Interval to send a KEEPALIVE packet. None will turn off KEEPALIVE.
     )
-    KEEPALIVE_DATA = b"sitting well"  # Data to send out as part of the KEEPALIVE packet. Should not be null nor empty.
+    KEEPALIVE_DATA = (
```
Comment (LeStarch) [inline Gds/src/fprime_gds/common/adapters/ip.py:49]: Why? This the format tool?

File: Gds/src/fprime_gds/common/models/common/channel_telemetry.py [hunk 8]
```diff
@@ -6,28 +6,27 @@
...
-
-from fprime.common.models.serialize.string_type import *
-from fprime.common.models.serialize.serializable_type import *
-
-import struct
+from fprime.common.models.serialize.type_exceptions import TypeException
+from fprime.common.models.serialize.type_exceptions import TypeMismatchException
+from fprime.common.models.serialize.type_base import BaseType
+
+# from fprime.common.models.serialize.bool_type import *
```
Comment (LeStarch) [inline Gds/src/fprime_gds/common/models/common/channel_telemetry.py:13]: Just delete, no need to comment out.

File: Gds/src/fprime_gds/common/templates/event_template.py [hunk 9]
```diff
@@ -114,7 +112,7 @@ def get_format_str(self):
         return self.format_str
 
     def get_description(self):
-        self.description
+        return self.description
```
Comment (LeStarch) [inline Gds/src/fprime_gds/common/templates/event_template.py:115]: Ouch....this seems like it was needed.

File: Gds/src/fprime_gds/executables/tcpserver.py [hunk 10]
```diff
@@ -548,7 +541,7 @@ def main(argv=None):
         server_thread.start()
         udp_server_thread.daemon = False
         udp_server_thread.start()
-        p = os.getpid()
+        # p = os.getpid()
```
Comment (LeStarch) [inline Gds/src/fprime_gds/executables/tcpserver.py:544]: Delete both these lines.

File: Gds/src/fprime_gds/flask/default_settings.py [hunk 11]
```diff
@@ -28,7 +28,7 @@
 # Gds config setup
 GDS_CONFIG = fprime_gds.common.utils.config_manager.ConfigManager()
 
-for dir in [LOG_DIR, UPLOADED_UPLINK_DEST, UPLOADS_DEFAULT_DEST, DOWNLINK_DIR]:
-    os.makedirs(dir, exist_ok=True)
+for directory in [LOG_DIR, UPLOADED_UPLINK_DEST, UPLOADS_DEFAULT_DEST, DOWNLINK_DIR]:
```
Comment (LeStarch) [inline Gds/src/fprime_gds/flask/default_settings.py:31]: Me??? Improper variable names???  #blame @LeStarch

File: Fw/Python/src/fprime/common/models/serialize/bool_type.py [hunk 12]
```diff
@@ -68,14 +68,4 @@ def __repr__(self):
 
 
 if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (hunterpaulson) [reply Fw/Python/src/fprime/common/models/serialize/bool_type.py:70]: yes I can do that

File: Fw/Python/src/fprime/fbuild/cmake.py (hunk 4 above)
Comment (hunterpaulson) [reply Fw/Python/src/fprime/fbuild/cmake.py:541]: Will do

File: Gds/src/fprime_gds/common/adapters/ip.py (hunk 7 above)
Comment (hunterpaulson) [reply Gds/src/fprime_gds/common/adapters/ip.py:49]: Yes the format tool regards the comment as an important part of that line so it formats to keep it 'on' the same line

File: Gds/src/fprime_gds/flask/default_settings.py (hunk 11 above)
Comment (hunterpaulson) [reply Gds/src/fprime_gds/flask/default_settings.py:31]: haha, without pylint stuff like this happens

File: Fw/Python/src/fprime/common/models/serialize/time_type.py (hunk 2 above)
Comment (hunterpaulson) [reply Fw/Python/src/fprime/common/models/serialize/time_type.py]: I was testing to see if it was needed, turns out it wasnt and I forgot to delete

File: Fw/Python/src/fprime/fbuild/cmake.py (hunk 3 above)
Comment (hunterpaulson) [reply Fw/Python/src/fprime/fbuild/cmake.py:463]: Sadly yes. Having a dictionary as the default value without a 'is none' can cause some weird errors

**Pull request merged**